Rename commands to buildtest buildspec edit-test and buildtest buildspec edit-file

JeffreyFG opened the pull request:
Changed the documentation to reflect the change to the ``buildtest buildspec edit-test`` command. The alias for the buildspec command that allowed you to edit tests based on a supplied test name has been changed from ``buildtest buildspec edit`` to ``buildtest buildspec edit-test``.  This was done in order to better match with the ``buildtest buildspec edit-file``.

JeffreyFG committed:
Update buildspecs_interface.rst

Changed the documentation to reflect the change to the ``buildtest buildspec edit-test`` command. The alias for the buildspec command that allowed you to edit tests based on a supplied test name has been changed from ``buildtest buildspec edit`` to ``buildtest buildspec edit-test``.  This was done in order to better match with the ``buildtest buildspec edit-file``.

shahzebsiddiqui commented on docs/gettingstarted/buildspecs_interface.rst:
```diff
@@ -421,12 +421,12 @@ Edit buildspecs ``buildtest buildspec edit-file``
 The ``buildtest buildspec edit-file`` command can be used to edit buildspec based on filename as pose to testname.
-This command works similar to ``buildtest buildspec edit`` where each file is open in editor and validated upon completion.
-You can use this command to create new buildspec whereas ``buildtest buildspec edit`` only works on existing buildspecs loaded
+This command works similar to ``buildtest buildspec edit-test`` where each file is open in editor and validated upon completion.
```
I think you mean `buildtest buildspec edit-file` here since this is section on editing by buildspecs. The section above is for `edit-test`

shahzebsiddiqui commented on docs/gettingstarted/buildspecs_interface.rst:
```diff
@@ -421,12 +421,12 @@ Edit buildspecs ``buildtest buildspec edit-file``
-This command works similar to ``buildtest buildspec edit`` where each file is open in editor and validated upon completion.
-You can use this command to create new buildspec whereas ``buildtest buildspec edit`` only works on existing buildspecs loaded
+This command works similar to ``buildtest buildspec edit-test`` where each file is open in editor and validated upon completion.
+You can use this command to create new buildspec whereas ``buildtest buildspec edit-test`` only works on existing buildspecs loaded
```
Same here please change to `edit-file`

JeffreyFG committed:
changed cli argument from edit to edit-test in bash_completion.sh

JeffreyFG committed:
changed parser tokens from edit to edit-test in __init__.py to accommodate name change

shahzebsiddiqui commented on bash_completion.sh:
```diff
@@ -83,7 +83,7 @@ _buildtest ()
   COMPREPLY=()   # Array variable storing the possible completions.
 
-  local cmds="build buildspec cd cdash clean config debugreport docs edit help inspect history path report schema schemadocs stylecheck unittests"
+  local cmds="build buildspec cd cdash clean config debugreport docs edit-test help inspect history path report schema schemadocs stylecheck unittests"
```
please remove `edit-test` from here these are the top level commands  such as `buildtest build`. The `buildtest buildspec edit-test` is a subcommand which is followed in separate case statement.

shahzebsiddiqui commented on bash_completion.sh:
```diff
@@ -158,7 +158,7 @@ _buildtest ()
       ;;
     config|cg)
-      local cmds="-h --help compilers edit executors validate view systems"
+      local cmds="-h --help compilers edit-test executors validate view systems"
```
This is the section for `buildtest config` and we have command `buildtest config edit` which is a valid command. Please revert change

shahzebsiddiqui commented on bash_completion.sh:
```diff
@@ -216,7 +216,7 @@ _buildtest ()
 
     buildspec|bc)
-      local cmds="-h --help edit edit-file find show summary validate"
+      local cmds="-h --help edit-test edit-file find show summary validate"
```
This is correct place to make change! Good job. 

You can test bash completion by doing `source bash_completion.sh` and then `TAB` away :-)

shahzebsiddiqui commented on buildtest/cli/__init__.py:
```diff
@@ -658,7 +658,7 @@ def config_menu(subparsers):
     compilers = subparsers_config.add_parser("compilers", help="Search compilers")
 
-    subparsers_config.add_parser("edit", help="Open configuration file in editor")
+    subparsers_config.add_parser("edit-test", help="Open configuration file in editor")
```
So this method `config_menu` is for building the arguments for `buildtest config` and we want to keep `buildtest config edit` command so please revert this change.

JeffreyFG committed:
Update bash_completion.sh

Fixed changes to edit commands that did not have to do with edit test or edit-file.

JeffreyFG committed:
Fixed tab completion and changed the name of the edit to edit via the way the file is being selected

JeffreyFG committed:
removed unused import
